DLPX-86142 [insight] Unknown command fluentd error while creating insight config

rasantel opened the pull request:
<details open>
<summary><h2> Problem </h2></summary>

After upgrading from 10 to 11 when there is an enabled fluentd configuration for a plugin such as [elasticsearch-7.far](https://gitlab.delphix.com/brad.lewis/far-dev/blob/master/elasticsearch-7.far), fluentd and the management stack fail to start.
</details>

<details open>
<summary><h2> Diagnosis </h2></summary>

The fluentd logs show a failure to find a dependency of `elasticsearch-7`, the gem `faraday` of major version 1; only version 3.x is available. It turns out that this older gem version, which used to be included in the `td-agent` package in 10.0 (version [4.4.2-1](https://s3.amazonaws.com/packages.treasuredata.com/4/ubuntu/focal/pool/contrib/t/td-agent/td-agent_4.4.2-1_amd64.deb)) is no longer available in 11.0 (version [4.5.0-1](https://s3.amazonaws.com/packages.treasuredata.com/4/ubuntu/focal/pool/contrib/t/td-agent/td-agent_4.5.0-1_amd64.deb)). The upgrade of `td-agent` removes that gem, among others. The fluentd container start script copies the gems from td-agent to a new directory, but the stop script deletes that directory, so the old gems are not available anymore.
</details>

<details open>
<summary><h2> Solution </h2></summary>

Until we decide how to deal with plugins that are missing dependencies after upgrading ([DLPX-86157](https://delphix.atlassian.net/browse/DLPX-86157)) and we let the stack start normally even if a plugin fails ([DLPX-86156](https://delphix.atlassian.net/browse/DLPX-86156)), we'll pin td-agent to its version in 10.0 (version [4.4.2-1](https://s3.amazonaws.com/packages.treasuredata.com/4/ubuntu/focal/pool/contrib/t/td-agent/td-agent_4.4.2-1_amd64.deb)).

Specifically, here on `linux-pkg`, we make sure that this older version of `td-agent` is available when building the appliance.

Companion app gate review that forces virtualization to depend on this older version: https://github.com/delphix/dlpx-app-gate/pull/728.
</details>


<details open>
<summary><h2> Testing Done </h2></summary>

Verified manually. In a 11.0 engine, reproduced the issue by uploading the [elasticsearch-7.far](https://gitlab.delphix.com/brad.lewis/far-dev/blob/master/elasticsearch-7.far) plugin and enabling it. Then, I downgraded `td-agent` to the older version and verified that fluentd and the stack start successfully.

In progress: appliance build with both changes at http://selfservice.jenkins.delphix.com/job/appliance-build-orchestrator-pre-push/5495/ . Will upload the plugin and enable it when it completes.

`git ab-pre-push -b misc-debs`: http://selfservice.jenkins.delphix.com/job/appliance-build-orchestrator-pre-push/5494/
</details>

prakashsurya reviewed (approved):
I haven't been involved in the RCA, so I can't say if this is a good way to fix the problem or not, but it seems reasonable otherwise.

Does this need to go into `release` as well?

prakashsurya commented:
Can you run `git-ab-pre-push -b misc-debs` and post a link to the run? we don't run this automatically for this repository.. so it'd be good to verify things..

rasantel commented:
> Does this need to go into `release` as well?

@prakashsurya I meant to target `release`. Fixed.

> Can you run `git-ab-pre-push -b misc-debs` and post a link to the run? we don't run this automatically for this repository.. so it'd be good to verify things..

Done. Added link to the testing section.

rasantel commented:
> Can you run `git-ab-pre-push -b misc-debs` and post a link to the run? we don't run this automatically for this repository.. so it'd be good to verify things..

prakashsurya reviewed (approved):
Thanks for the details in the description, given those, this change makes sense to me.

Could you also put some information w.r.t. where the debian package that we put into artifactory came from? Just in case we need that information later.

E.g. did we copy it from our Ubuntu mirror, and place it in artifactory (no further modifications)? Did we build it from a source package? Something else?

rasantel commented:
>Could you also put some information w.r.t. where the debian package that we put into artifactory came from? Just in case we need that information later.
> E.g. did we copy it from our Ubuntu mirror, and place it in artifactory (no further modifications)? Did we build it from a source package? Something else? 

Done

rasantel committed:
DLPX-86063 Management service stuck in activating state after deferred upgrade release(10) to develop(12)

PR URL: https://www.github.com/delphix/linux-pkg/pull/288
